src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsworkload/integration/listener/WorkloadCalculationListenerTests.kt

```diff
@@ -0,0 +1,57 @@
+package uk.gov.justice.digital.hmpps.hmppsworkload.integration.listener
+
+import com.amazonaws.services.sns.model.MessageAttributeValue
+import com.amazonaws.services.sns.model.PublishRequest
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.CaseType
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.Tier
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.event.PersonReference
+import uk.gov.justice.digital.hmpps.hmppsworkload.domain.event.PersonReferenceType
+import uk.gov.justice.digital.hmpps.hmppsworkload.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.CaseDetailsEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.PersonManagerEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.jpa.entity.WorkloadCalculationEntity
+import uk.gov.justice.digital.hmpps.hmppsworkload.listener.AdditionalInformation
+import uk.gov.justice.digital.hmpps.hmppsworkload.listener.WorkloadCalculationEvent
+import java.math.BigDecimal
+import java.math.BigInteger
+import java.time.LocalDateTime
+
+class WorkloadCalculationListenerTests : IntegrationTestBase() {
+
+  @Test
+  fun `calculate workload when staff available hours have changed`() {
+    val crn = "J678910"
+    val staffCode = "staff1"
+    val teamCode = "team1"
+
+    val caseDetailsEntity = CaseDetailsEntity(crn, Tier.C3, CaseType.COMMUNITY, "Jane", "Doe")
+
+    caseDetailsRepository.save(caseDetailsEntity)
+
+    staffCodeResponse(staffCode, teamCode)
+    personManagerRepository.save(PersonManagerEntity(crn = crn, staffId = BigInteger.ONE, staffCode = staffCode, teamCode = teamCode, offenderName = "offender", createdBy = "createdby", providerCode = "providerCode", isActive = true))
+
+    hmppsDomainSnsClient.publish(
+      PublishRequest(hmppsDomainTopicArn, jsonString(staffAvailableHoursChangedEvent(staffCode, teamCode, BigDecimal.TEN))).withMessageAttributes(
+        mapOf("eventType" to MessageAttributeValue().withDataType("String").withStringValue("staff.available.hours.changed"))
+      )
+    )
+
+    noMessagesOnWorkloadCalculationEventsQueue()
+
+    val actualWorkloadCalcEntity: WorkloadCalculationEntity? =
+      workloadCalculationRepository.findFirstByStaffCodeAndTeamCodeOrderByCalculatedDate(staffCode, teamCode)
+
+    Assertions.assertAll(
```

Comment:
I might be missing something but shouldn't we check the new available hours? That is ultimately what we are testing here?!